web/src/message_list_view.js

```diff
@@ -1576,4 +1577,23 @@ export class MessageListView {
             stream_color.update_stream_recipient_color($stream_header);
         }
     }
+
+    show_message_as_read(message, options) {
+        const $row = this.get_row(message.id);
+        if (options.from === "pointer" || options.from === "server") {
+            $row.find(".unread_marker").addClass("fast_fade");
+        } else {
+            $row.find(".unread_marker").addClass("slow_fade");
+        }
+        $row.removeClass("unread");
+    }
+
+    show_messages_as_unread(message_ids) {
+        const $table = rows.get_table(this.table_name);
+        const $rows_to_show_as_unread = $table.find(".message_row").filter((index, $row) => {
```

Comment:
I switched this to use `filter(index, $row)` to save an extra lookup that someone might feel the need to think about whether it can fail.